acra/configuring-maintaining/optimisations/acrastructs_vs_acrablocks.md

```diff
@@ -0,0 +1,44 @@
+---
+title: AcraStructs vs AcraBlocks
+weight: 3
+---
+
+# AcraStructs vs AcraBlocks
+
+There are two different "crypto containers" AcraServer can use to store encrypted value in.
+Then, these containers are stored in database and decrypted back by AcraServer when needed.
+
+`AcraStruct` is the main crypto container for the client-side encryption.
+Applications can encrypt with ephemeral asymmetric keys on their side with `AcraStruct`s,
+but cannot do this with `AcraBlock`s because of symmetric key encryption.
```

Comment:
This reads like blaming AcraBlocks for not doing what they were explicitly designed not to do.

```suggestion
AcraStruct is the main crypto container for the client-side encryption.
AcraStructs use ephemeral asymmetric keys to enable applications to encrypt the data but never decrypt it –
the private key stays on the AcraServer which is the only one capable of decryption.
```

And you can add a blurb about AcraBlocks in a separate paragraph after this one. For example,

```markdown
AcraBlocks use only symmetric cryptography to they are generally faster then AcraStructs.
However, this comes at a cost of having to do stricter key management,
since now the application has the decryption keys as well which is a risk.
```